### PR TITLE
Modules 01: move files to start the modularization of the codebase

### PR DESCRIPTION
**Pull request by bakulf**
This is the first PR for the modularization of the codebase. All is moved in `src/apps/vpn` except for `mbft`, `rfc` and `hacl-start`, the only 3 components without external dependencies.

### CODE REVIEW

File: nebula/ui/nebula_resources.qrc [hunk 1]
```diff
@@ -307,7 +307,6 @@
         <file>resources/Icon1024.png</file>
         <file>resources/location-dark.svg</file>
         <file>resources/logo-dock.png</file>
-        <file>resources/logo.ico</file>
```
Comment (brizental) [inline nebula/ui/nebula_resources.qrc:310]: How is this change related?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -21,10 +21,11 @@ if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Emscripten"
     )
 endif()
 
-include(cmake/sources.cmake)
-include(cmake/sentry.cmake)
+include(shared/sources.cmake)
+include(apps/vpn/cmake/sources.cmake)
```
Comment (brizental) [inline src/CMakeLists.txt:25]: Why to we have all of these vpn includes inside the src/CMakeLists.txt file? Should there be a vpn specific CMakeLists doing all this?

File: src/apps/vpn/cmake/android.cmake [hunk 3]
```diff
@@ -0,0 +1,86 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (brizental) [inline src/apps/vpn/cmake/android.cmake:1]: This is just a copied file right? How annoying would it be to fix the git history so that we know it is moved and not written from scratch?

File: src/apps/vpn/cmake/linux.cmake [hunk 4]
```diff
@@ -0,0 +1,133 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (brizental) [inline src/apps/vpn/cmake/linux.cmake:1]: Same comment as for android.cmake.

File: src/apps/vpn/cmake/wasm.cmake [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (brizental) [inline src/apps/vpn/cmake/wasm.cmake:1]: Same.

File: tests/qml/tst_mainWindowGleanEvents.qml [hunk 6]
```diff
@@ -1,3 +1,7 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
[the remark this answers is not in the document]
Comment (brizental) [reply tests/qml/tst_mainWindowGleanEvents.qml:3]: Filed a ticket: https://mozilla-hub.atlassian.net/browse/VPN-3515

**Review by brizental — CHANGES_REQUESTED** [review]
I see this moves files and folders under src/, but what about the root folders and files that are also specific to the vpn app? From a quick look I see for example Android daemon stuff is vpn specific.

File: nebula/ui/nebula_resources.qrc (hunk 1 above)
Comment (bakulf) [reply nebula/ui/nebula_resources.qrc:310]: this icon is totally not used. It's a dup of what we have in apps/vpn/ui/resources/logo.ico.

File: src/CMakeLists.txt (hunk 2 above)
Comment (bakulf) [reply src/CMakeLists.txt:25]: because this PR is just about moving files. I'm not changing the build system as we/I like to.

File: src/apps/vpn/cmake/android.cmake (hunk 3 above)
Comment (bakulf) [reply src/apps/vpn/cmake/android.cmake:1]: yes, plus path changes -> src/platfroms/android -> src/apps/vpn/platforms/android

File: src/apps/vpn/cmake/linux.cmake (hunk 4 above)
Comment (bakulf) [reply src/apps/vpn/cmake/linux.cmake:1]: See the previous comment

File: src/apps/vpn/cmake/wasm.cmake (hunk 5 above)
Comment (bakulf) [reply src/apps/vpn/cmake/wasm.cmake:1]: See the previous comment

File: src/apps/vpn/cmake/windows.cmake [hunk 7]
```diff
@@ -0,0 +1,133 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
```
[the remark this answers is not in the document]
Comment (bakulf) [reply src/apps/vpn/cmake/windows.cmake:1]: See the previous comment

File: src/apps/vpn/networkmanager.cpp [hunk 8]
```diff
@@ -6,7 +6,6 @@
 
 #include "constants.h"
 #include "leakdetector.h"
-#include "models/feature.h"
```
[the remark this answers is not in the document]
Comment (bakulf) [reply src/apps/vpn/networkmanager.cpp:9]: Moving files sometimes I saw wrong/unused includes and... I fixed them.

File: tests/qml/tst_mainWindowGleanEvents.qml (hunk 6 above)
[the remark this answers is not in the document]
Comment (bakulf) [reply tests/qml/tst_mainWindowGleanEvents.qml:3]: Thanks!

**Review by lesleyjanenorton — APPROVED** [review]

File: src/apps/vpn/platforms/windows/daemon/windowsfirewall.cpp [hunk 9]
```diff
@@ -21,9 +21,8 @@
 #include <QScopeGuard>
 #include <QtEndian>
 
-#include "../../daemon/interfaceconfig.h"
```
Comment (Gela) [inline src/apps/vpn/platforms/windows/daemon/windowsfirewall.cpp:24]: Just to confirm, was this removal intentional?

**Review by Gela — APPROVED** [review]

File: src/apps/vpn/platforms/windows/daemon/windowsfirewall.cpp (hunk 9 above)
Comment (bakulf) [reply src/apps/vpn/platforms/windows/daemon/windowsfirewall.cpp:24]: yes. it was already included in windowsfirewall.h

File: src/apps/vpn/platforms/windows/daemon/windowsfirewall.h [hunk 10]
```diff
@@ -18,7 +18,7 @@
 #include <QObject>
 #include <QString>
 
-#include "../../daemon/interfaceconfig.h"
+#include "daemon/interfaceconfig.h"
```
[the remark this answers is not in the document]
Comment (bakulf) [reply src/apps/vpn/platforms/windows/daemon/windowsfirewall.h:21]: nope. It compiles without issues.

[the remark this answers is not in the document]
Comment (bakulf) [reply src/apps/vpn/platforms/windows/daemon/windowsfirewall.h:21]: Plus I fixed the include paths
